### PR TITLE
#25 added --cached option for clean command

### PR DESCRIPTION
**Pull request by maxonfjvipon**

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@yegor256 please, check this out

**Comment by maxonfjvipon** [comment]
@yegor256 please, check this out, should work

**Comment by maxonfjvipon** [comment]
@yegor256 please check this out, should work

**Comment by yegor256** [comment]
@maxonfjvipon the problem here is that `--cached` will be available for _all_ commands, but we want it to be available _only_ for `clean`. In other words, `--cached` should not print in the default list of options. It should print only when I say:

```
$ eoc clean --help
```

**Comment by maxonfjvipon** [comment]
@yegor256 got you

**Comment by maxonfjvipon** [comment]
@yegor256 I think that's it

File: test/commands/test_clean.js [hunk 1]
```diff
@@ -25,18 +25,22 @@
...
+    fs.rmSync(eo, {recursive: true, force: true});
     fs.mkdirSync(path.resolve(home, 'src'), {recursive: true});
+    fs.mkdirSync(eo, {recursive: true});
     fs.writeFileSync(path.resolve(home, 'src/simple.eo'), '[] > simple\n');
     const stdout = runSync([
-      'clean', '-s', path.resolve(home, 'src'), '-t', path.resolve(home, 'target'),
+      'clean', '-s', path.resolve(home, 'src'), '-t', path.resolve(home, 'target'), '--cached',
     ]);
-    assert(!fs.existsSync(path.resolve(home, 'target')), stdout);
+    assert(!fs.existsSync(path.resolve(home, 'target')) && !fs.existsSync(eo), stdout);
```
Comment (yegor256) [inline test/commands/test_clean.js:43]: @maxonfjvipon if this assertion fails, it won't be possible to understand what exactly was the problem: the `target` is absent or `~/.eo`? It's better to have individual `assert` statements per each particular error you are checking.

File: src/eoc.js [hunk 2]
```diff
@@ -69,10 +69,12 @@ program.command('foreign')
...
   });
 
-program.command('clean')
+program
+  .command('clean')
+  .option('--cached', 'delete ~/.eo directory')
   .description('delete all temporary files')
   .action((str, opts) => {
-    clean(program.opts());
+    clean({...str, ...program.opts()});
```
Comment (yegor256) [inline src/eoc.js:77]: @maxonfjvipon I have a strong feeling that it's wrong. Try to run this and you will the problem:

```
$ eoc clean --cached --foo
```

**Comment by yegor256** [comment]
@maxonfjvipon see above

File: src/eoc.js (hunk 2 above)
Comment (maxonfjvipon) [reply src/eoc.js:77]: with `--foo` we get `error: unknown option '--foo'`, so it works absolutely correctly, but not quite obviously, will fix it.

File: test/commands/test_clean.js (hunk 1 above)
Comment (maxonfjvipon) [reply test/commands/test_clean.js]: fixed

File: src/eoc.js (hunk 2 above)
Comment (maxonfjvipon) [reply src/eoc.js]: fixed

**Comment by maxonfjvipon** [comment]
@yegor256 please check one more time

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@maxonfjvipon did you try `grunt eslint`?

**Comment by maxonfjvipon** [comment]
@yegor256 please try merge one more time

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**